### ISSUE TITLE
Persistent Error: Failed to read data from storage

### ISSUE THREAD

**Issue by Cynicath** [MutinyWallet/mutiny-web#529]
Wallet stopped working on Android app with the above error. Tried saving and importing state via the app and web wallet. Tried uninstalling app and reinstalling. No luck.

**Comment by TonyGiorgio** [comment]
Thanks for making an issue. Can you download the logs and upload them here?

**Comment by Cynicath** [comment]
[mutiny-logs.txt](https://github.com/MutinyWallet/mutiny-web/files/12603263/mutiny-logs.txt)

**Comment by benthecarman** [comment]
Normally this is fixed by restoring from seed rather than importing state

**Comment by Cynicath** [comment]
> Normally this is fixed by restoring from seed rather than importing state

I tried that on the app _and_ web browser. The problem persisted on both platforms.

**Comment by TonyGiorgio** [comment]
Do you happen to know if the logs you updated were the results from the state importing or from the seed recovery?

**Comment by Cynicath** [comment]
> Do you happen to know if the logs you uploaded were the results from the state importing or from the seed recovery?

I recovered the seed and then tried importing the state... so both? I can do one or the other if that would provide more info.

**Comment by TonyGiorgio** [comment]
Yes please. Click the "Delete Wallet" button and then recover with the seed and provide the logs. Don't upload the state file.

**Comment by Cynicath** [comment]
[mutiny-logs(1).txt](https://github.com/MutinyWallet/mutiny-web/files/12603344/mutiny-logs.1.txt)

**Comment by TonyGiorgio** [comment]
> Successfully loaded channel at update_id 3808 against monitor at update id 3807
2023-09-14 02:53:12.629 ERROR [lightning::ln::channelmanager:8653] A ChannelMonitor is stale compared to the current ChannelManager! This indicates a potentially-critical violation of the chain::Watch API!

This is what I was afraid of, this is the first time it's happened and it's very rare that it was possible. We'll have to get back to you on this tomorrow after we talk to the LDK team who is in town.

**Comment by Cynicath** [comment]
OK thanks for the speedy responses thus far, I appreciate it!

**Comment by TonyGiorgio** [comment]
I think there's something we can do but will have to build in something new to the emergency screen to handle this case. In a few weeks, a new version of LDK should be released and we can avoid this issue completely but it's super rare for this to happen if you could give us a bit to figure out what exactly to do, code it up, put it onto the emergency screen, and then get back to you for instructions.

**Comment by Cynicath** [comment]
Wow thank you for taking the time to work on a fix specifically for me even though preventative measures will be available soon, that's some incredible customer service. I'll be sure to recommend your wallet every chance I get. 

Is it safe for me to start another wallet so I can get back to zapping people, or is it likely that something in my setup caused the problem and could occur again?

Thanks again for the fast responses and awesome support.

**Comment by TonyGiorgio** [comment]
No problem at all, just sorry that happened to you. Appreciate your patience and kind words!

**Comment by Cynicath** [comment]
> No problem at all, just sorry that happened to you. Appreciate your patience and kind words!

Just wondering if there's been any progress on the issue. Should I just write off the sats as lost? Thanks again for looking into it.

**Comment by Cynicath** [comment]
Just wondering if there's been any progress on the issue. Should I just write off the sats as lost? Thanks again for looking into it.

**Comment by benthecarman** [comment]
Hey, we made a simple CLI tool that should help try to recover your funds. You should be able to download it and run `mutiny-data-recovery --seed "your seed here"` and make it so you can open mutiny again and hopefully get your funds back

https://github.com/MutinyWallet/mutiny-data-recovery/releases/tag/v0.1.0

**Comment by Cynicath** [comment]
> Hey, we made a simple CLI tool that should help try to recover your funds. You should be able to download it and run `mutiny-data-recovery --seed "your seed here"` and make it so you can open mutiny again and hopefully get your funds back
> 
> https://github.com/MutinyWallet/mutiny-data-recovery/releases/tag/v0.1.0

Thank you for this, much appreciated. Unfortunately I am not tech savvy enough to figure out how to use these files with the information given. I don't expect you to hold my hand but perhaps you could give me some search terms to Google so I can teach myself what to do? Preferably a resource that helps me obtain the knowledge to understand what I'm doing rather than just plugging in commands (though that would suffice). I'm using Ubuntu and I'm obviously a bit of a noob. I also have a Windows machine I tried using the .exe file on but nothing happened. If you don't have time to help I understand, it's my fault for putting funds into beta software that I don't have the skills to debug. Thanks either way for taking the time to code these files.

**Comment by Cynicath** [comment]
It occurs to me that I should probably give some info on where exactly I'm stuck. 

 cargo install mutiny-data-recovery
    Updating crates.io index
  Installing mutiny-data-recovery v0.1.0
error: failed to compile `mutiny-data-recovery v0.1.0`, intermediate artifacts can be found at `/tmp/cargo-installEKUT5j`

Caused by:
  package `clap_builder v4.4.7` cannot be built because it requires rustc 1.70.0 or newer, while the currently active rustc version is 1.66.1
  Try re-running cargo install with `--locked`

When I check the version with rustc -V it does indeed show that I am using version 1.66.1, so I tried updating the version via rustup. Then:

$ rustup show
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/g/snap/rustup/common/rustup

stable-x86_64-unknown-linux-gnu (directory override for '/home/g')
rustc 1.73.0 (cc66ad468 2023-10-03)

So it appears that I _have_ the current version of rustc but it isn't active?

I tried $ rustup default stable but that doesn't seem to fix it. 

I know there is something fundamental and simple that I'm missing but I don't know what it is. Again, I understand if you don't have time to troubleshoot for me.
